Source/WebCore/rendering/BreakLines.h

```diff
@@ -178,48 +182,57 @@ inline size_t BreakLines::nextBreakablePosition(CachedLineBreakIteratorFactory&
                 continue;
             }
 
+            // When nonBreakingSpaceBehavior == NoBreakSpaceBehavior::Break is specified, NBSP should behave as if it is just a space.
+            CharacterType beforeCharacter = before;
+            if constexpr (nonBreakingSpaceBehavior == NoBreakSpaceBehavior::Break) {
+                if (beforeCharacter == noBreakSpace)
+                    beforeCharacter = space;
+            }
+
             // If both characters are ASCII, use a lookup table for enhanced speed
             // and for compatibility with other browsers (see comments on lineBreakTable for details).
-            if (before <= lineBreakTable.lastCharacter && after <= lineBreakTable.lastCharacter) {
-                if (before >= lineBreakTable.firstCharacter && after >= lineBreakTable.firstCharacter) {
-                    if (lineBreakTable.unsafeLookup(before, after))
+            if (beforeCharacter <= lineBreakTable.lastCharacter && after <= lineBreakTable.lastCharacter) {
+                if (beforeCharacter >= lineBreakTable.firstCharacter && after >= lineBreakTable.firstCharacter) {
+                    if (lineBreakTable.unsafeLookup(beforeCharacter, after))
                         return i;
                 } // Else at least one is an ASCII control character; don't break.
                 continue;
             }
         }
 
         // Non-ASCII rapid lookup.
-        if (words != WordBreakBehavior::AutoPhrase) {
+        if constexpr (words != WordBreakBehavior::AutoPhrase) {
             if (!before.type)
                 before.type = classify<shortcutRules, nonBreakingSpaceBehavior>(before);
             after.type = classify<shortcutRules, nonBreakingSpaceBehavior>(after);
             // Short-circuit the commonest cases: letter + letter.
             int pair = before.type | after.type;
             if (pair == (kAL | kAL)) {
-                if (words == WordBreakBehavior::BreakAll)
+                if constexpr (words == WordBreakBehavior::BreakAll)
                     return i;
                 continue;
             }
             if ((pair | kAL) == (kID | kAL)) {
-                if (words == WordBreakBehavior::KeepAll)
+                if constexpr (words == WordBreakBehavior::KeepAll)
                     continue;
                 return i;
             }
             // Handle special cases.
-            if (pair & kGL && !(pair & kWeird)) // Keep nbsp high in our list.
+            if (pair & (kGL | kQU) && !(pair & kWeird)) // Keep nbsp high in our list.
```

Comment:
This breaks the handling of NBSP sequences and passes them to ICU. I don't think we want to do that. :) That would be a significant perf regression on pages that have extensive use of nbsp (which we have specific optimizations checked in for in the past).